packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -92,6 +92,10 @@ class BoxDecoration extends Decoration {
     this.boxShadow,
     this.gradient,
     this.backgroundBlendMode,
+    @Deprecated(
+      'Replace BoxDecoration(shape: BoxShape.circle) with ShapeDecoration(shape: const CircleBorder() '
```

Comment:
This should probably be more generic, some users may not be using circle.

Reply:
BoxShape.rectangle doesn't do anything. I guess, "If BoxShape.rectangle, don't do anything. If BoxShape.circle, replace with ShapeDecoration"?

Reply:
yeah the default is rectangle, and the only other option is circle.